packages/shorebird_cli/test/src/validators/tracked_lock_files_validator_test.dart

```diff
@@ -0,0 +1,153 @@
+    });
+
+    setUp(() {
+      git = MockGit();
+      shorebirdEnv = MockShorebirdEnv();
+      projectRoot = Directory.systemTemp.createTempSync();
+
+      when(() => shorebirdEnv.getFlutterProjectRoot()).thenReturn(projectRoot);
+
+      validator = TrackedLockFilesValidator();
+    });
+
+    test('has a non-empty description', () {
+      expect(validator.description, isNotEmpty);
+    });
+
+    group('canRunInCurrentContext', () {
+      group('when a pubspec.yaml file exists', () {
+        setUp(() {
+          when(() => shorebirdEnv.hasPubspecYaml).thenReturn(true);
+        });
+
+        test('returns true', () {
+          expect(runWithOverrides(validator.canRunInCurrentContext), isTrue);
+        });
+      });
+
+      group('when a pubspec.yaml file does not exist', () {
+        setUp(() {
+          when(() => shorebirdEnv.hasPubspecYaml).thenReturn(false);
+        });
+
+        test('returns false', () {
+          expect(runWithOverrides(validator.canRunInCurrentContext), isFalse);
+        });
+      });
+    });
+
+    group('validate', () {
+      group('when no project root is found', () {
+        setUp(() {
+          when(() => shorebirdEnv.getFlutterProjectRoot()).thenReturn(null);
+        });
+
+        test('returns an empty list', () async {
+          final issues = await runWithOverrides(() => validator.validate());
+          expect(issues, isEmpty);
+        });
+      });
+
+      group('when project is not tracked in git', () {
+        setUp(() {
+          when(
+            () => git.isGitRepo(directory: any(named: 'directory')),
+          ).thenAnswer((_) async => false);
+        });
+
+        test('returns no issues', () async {
+          final issues = await runWithOverrides(() => validator.validate());
+          expect(issues, isEmpty);
+        });
+      });
+
+      group('when a lock file does not exist', () {
+        setUp(() {
+          when(
+            () => git.isGitRepo(directory: any(named: 'directory')),
+          ).thenAnswer((_) async => true);
+        });
+
+        test('does not warn about lock file not being tracked', () async {
+          final issues = await runWithOverrides(() => validator.validate());
+          expect(issues, isEmpty);
+        });
+      });
+
+      group('when a lock file exists but is not tracked', () {
+        setUp(() {
+          when(
+            () => git.isGitRepo(directory: any(named: 'directory')),
+          ).thenAnswer((_) async => true);
+          when(
+            () => git.isFileTracked(file: any(named: 'file')),
+          ).thenAnswer((_) async => false);
+
+          File(p.join(projectRoot.path, 'pubspec.lock')).createSync();
+        });
+
+        test('recommends adding lock file to source control', () async {
+          final issues = await runWithOverrides(() => validator.validate());
+          expect(issues, hasLength(1));
+          expect(
+            issues.first.message,
```

Comment:
should we also assert that the severity is warning?

Reply:
Done